File: src/contexts/chat/domain/value-objects/finish-reason/finish-reason.value-object.ts
import { FinishReasonEnum } from '@contexts/chat/domain/enums/finish-reason.enum';
import { EnumValueObject } from '@sisques-labs/nestjs-kit';

/**
 * Why the model stopped generating text.
 */
export class FinishReasonValueObject extends EnumValueObject<
  typeof FinishReasonEnum
> {
  constructor(value: FinishReasonEnum) {
    super(value);
  }

  protected get enumObject(): typeof FinishReasonEnum {
    return FinishReasonEnum as unknown as typeof FinishReasonEnum;
  }
}

File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/domain/valueobjects/FinishReasonValueObject.java
package com.sisqueslabs.nexora.api.contexts.chat.domain.valueobjects;

import com.fasterxml.jackson.annotation.JsonValue;

/**
 * Why the model stopped generating text.
 */
public enum FinishReasonValueObject {
    STOP("stop");

    private final String wireValue;

    FinishReasonValueObject(String wireValue) {
        this.wireValue = wireValue;
    }

    @JsonValue
    public String wireValue() {
        return wireValue;
    }
}
